services/can_logger/can_logger.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python3
+
+import socket
+import postgres
+import csv
+import os
+
+def csv_init(host_interface):
+
+    log = open('/data/log/' + host_interface + '.csv', mode = 'a') 
+
+    return log
+
+def write_to_csv(timestamp, can_id, can_data, log):
+
+        log = csv.writer(log, delimiter = ',', quotechar = '"',
+                         quoting = csv.QUOTE_MINIMAL)
+
+        log.writerow([timestamp, can_id, can_data])
+
+def write_to_db(rx_buff, db):
+
+    db.run("INSERT INTO can (time, can_id, can_data) VALUES (%s)", (rx_buff))
+
+def db_init():
+    
+    connection_url = ('postgresql://' + os.environ['db_user'] + ':' + 
+                 os.environ['db_password'] + '@postgres:' + 
+                 os.environ['db_port'] + '/' + os.environ['db_database'] )
+
+    print('Initializing Postgres Object...')
+    db = postgres.Postgres(url = connection_url)
+
+    print('Ensuring timescaledb ext. is enabled')
+    db.run("CREATE EXTENSION IF NOT EXISTS timescaledb;")
+    print("Ensuring tables are setup properly")
+    db.run("""
+           CREATE TABLE IF NOT EXISTS can (
+               time timestamptz UNIQUE NOT NULL,
+               can_id text NOT NULL,
+               can_data text NOT NULL);""")
+
+    print("Ensuring can data table is a timescaledb hypertable")
+    db.run("""
+           SELECT create_hypertable('can', 'time', if_not_exists => TRUE,  
+           migrate_data => TRUE);""")
+
+    print("Finished setting up tables")
+
+    return db
+
+# Get host info using environment variables
+
+host_ip = os.environ['socketcand_ip']
+host_port = os.environ['socketcand_port']
+host_interface = os.environ['can_interface']
+logging = os.environ['log']
+
+# Initialize received frame buffer
+
+rx_buf = []
+
+# Initialize postgres database if database logging is enabled
+
+if(logging.find('db') != -1):
+    
+    db = db_init()
+
+# Initialize CSV file if CSV logging is enabled
+
+if(logging.find('csv') != -1):
+
+    csv_log = csv_init()
```

Comment:
Another nitpick, but it might be better to do something like this:

```suggestion
logtodb = False
logtocsv = False
if logging.find('db') != -1:
    logtodb = True
if logging.find('csv') != -1:
    logtocsv = True

# Initialize postgres database if database logging is enabled
if logtodb:
    db = db_init()

# Initialize CSV file if CSV logging is enabled
if logtocsv:
    csv_log = csv_init()
```

This way if we ever choose to change how to determine if we are doing csv or db logging, it's only one part of the code to change, and we can just check a boolean value instead of calling `.find()`, which might make a difference in the time sensitive `while(true)` loop below.